### PR TITLE
Create 2017-05-05-MM-DD-Updated-Security-Hardening-For-Sitecore-8.2.md

### PR DESCRIPTION
**Pull request by grant-killian**
In a comment on the pull request, include the following information and check off the list.

 - [x] Mini bio
  - Soft capped to 75 words
 - [x] Desired date of publishing
  - Make sure this matches the `date` in the front matter
 - [x] Social media contact info
 - [x] Redaction: (yes|no)
  - Allow redaction, editing or structure changes by technical writer staff
 - [x] Comments: (yes|no)
  - If you're willing to answer comments about your post, set `comments: true`.
  - Make sure to subscribe to your post comments by clicking on the **Subscribe** link at the very bottom of the post _once_ it's published.

@catlook Please review.

### CODE REVIEW

**Comment by rackernexus** [comment]
:mag_right: [Preview](https://staging.developer.rackspace.com/build-7191cd603c/)

**Comment by grant-killian** [comment]
Mini bio: Grant Killian is a Sitecore Architect on the Rackspace Managed Services for Sitecore team; his team helps enterprise customers be successful with the Sitecore platform
Desired date of publish: 5/5/2017
Social Media link: https://twitter.com/SitecoreAgent
Redaction: yes -- but some terminology may be Sitecore specific I may redact the redactions :)
@catlook Please review

**Comment by grant-killian** [comment]
Comments: No

File: _posts/2017-05-05-MM-DD-Updated-Security-Hardening-For-Sitecore-8.2.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+New Recommendation Number 4. [Change the security hash algorithm](https://doc.sitecore.net/sitecore_experience_platform/setting_up_and_maintaining/security_hardening/configuring/change_the_hash_algorithm_for_password_encryption)
+
+> Sitecore is an extensible platform and the ability to change which hashing algorithm is used by the out-of-the-box .Net Membership provider has been around for years.  There's a good write-up from last year on [replacing the hashing algorithm](http://blog.baslijten.com/sitecore-security-1-how-to-replace-the-password-hashing-algorithm/) and even back in 2013 [the concept was being explored](https://kamsar.net/index.php/2013/09/upgrading-sitecores-password-security/).  Just recently, however, Sitecore has published the *recommendation* to do this in an acknowledgement that SHA1 isn't going to cut it in 2017.  In fact, Sitecore now states "you must change the weak default hash algorithm (SHA1) that is used to encrypt user passwords to a stronger algorithm" and it doesn't get more clear than that!
```
Comment (catlook) [inline _posts/2017-05-05-MM-DD-Updated-Security-Hardening-For-Sitecore-8.2.md]: ... platform, and the ...
... and even from back ...
... algorithm", and it ...

File: _posts/2017-05-05-MM-DD-Updated-Security-Hardening-For-Sitecore-8.2.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	"@
+	Write-Host $caveat -ForegroundColor DarkYellow
+
+While it's straight-forward to change the *hashAlgorithmType* attribute to use SHA512, it can be a pain to convert old user passwords to the new format.  The caveat in the above PowerShell points to [a sample page that could be used to reset a Sitecore password](https://gist.github.com/grant-killian/d2c17ec90adc4b1b99b6089172268571).  When standing up a new Sitecore environment and employing the security hardening steps included in this write-up, placing that file temporarily on the server and obtaining a new password through it can be part of the workflow.  Be sure to remove that utility page when you're done, though!
+
+I should mention, there are more sophisticated ways of resetting and obtaining a Sitecore password, but they exceed the scope of this piece focussing on Sitecore's new security hardening recommendations.  There are SQL Server stored procs to reset a password, and that could be folded into a process that updated the .Net Membership hashing algorithm etc.  I'll leave that as an exercise for the reader, should they want to pursue this further . . .
```
Comment (catlook) [inline _posts/2017-05-05-MM-DD-Updated-Security-Hardening-For-Sitecore-8.2.md]: ... a password, which could be folded into a process that updates the ...

File: _posts/2017-05-05-MM-DD-Updated-Security-Hardening-For-Sitecore-8.2.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+To conclude, I'll direct you to a unified PowerShell script that combines [our previous security hardening guidance](https://developer.rackspace.com/blog/Security-Hardening-for-Sitecore-Environments/) with the measures covered in this write-up.  The [full PowerShell is on GitHub and can help you jumpstart this process in your own projects](https://gist.github.com/grant-killian/a6b00ccbfe28b40b76181fbb369f5c02).  
+
+Finally, I'll point out that these security best practices are just a good starting point for a secure Sitecore implementation.  Our [Rackspace Sitecore team](https://www.rackspace.com/digital/sitecore) engages with customers on their particular situations and helps make them successful with the Sitecore platform; when it comes to security, one size doesn't fit all, but it's nice to have a solid foundation to build on.
```
Comment (catlook) [inline _posts/2017-05-05-MM-DD-Updated-Security-Hardening-For-Sitecore-8.2.md]: Finally, remember that ...
... Sitecore platform. When it comes ...

**Comment by grant-killian** [comment]
Thanks Cat; how do I “approve” the changes?  The option is disabled to me in the “Review Chagnes” dropdown list of actions.

**Comment by catlook** [comment]
Just make the edits and push it to your PR.  That should automatically clear them and make the article ready to go ☺

-- Cat Lookabaugh, Sr. Information Developer
       Technical and Engineering Services
Rackspace Hosting – the #1 Managed Cloud Company
Info Dev Team      cat.lookabaugh@rackspace.com
Office: (210) 312-8193   Cell: (210) 364-1060

**Comment by rackernexus** [comment]
:mag_right: [Preview](https://staging.developer.rackspace.com/build-ab02288aef/)

**Pull request merged**